sdk/metrics/src/jmhBasedTest/java/io/opentelemetry/sdk/metrics/internal/state/AsynchronousCounterMemoryUsageBenchmark.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.metrics.internal.state;
+
+import static org.assertj.core.api.Fail.fail;
+
+import io.opentelemetry.sdk.metrics.data.AggregationTemporality;
+import io.opentelemetry.sdk.metrics.export.MemoryMode;
+import java.lang.management.ManagementFactory;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CancellationException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * Benchmarks the memory usage of different memory modes and aggregation temporalities
+ *
+ * <p>This benchmark helps to see the difference in heap usage between the memory modes. You can
+ * change the aggregation temporality, number of counters and cardinality (attribute sets count).
+ * You have to run it manually for each parameter, since it's not JMH-based, and we need a clean
+ * heap before we start.
+ *
+ * <p>This benchmark class has additional usage: You can use it to see memory allocation frame
+ * graphs for a single run.
+ *
+ * <p>Steps:
+ *
+ * <ol>
+ *   <li>Follow download instructions for async-profiler, located at
+ *       https://github.com/async-profiler/async-profiler
+ *   <li>Assuming you have extracted it at /tmp/async-profiler-2.9-macos, add the following to your
+ *       JVM arguments of your run configuration:
+ *       <pre>
+ *       -agentpath:/tmp/async-profiler-2.9-macos/build/libasyncProfiler.so=start,event=alloc,flamegraph,file=/tmp/profiled_data.html
+ *       </pre>
+ *   <li>Tune the parameters as you see fit
+ *   <li>Run the class
+ *   <li>Open /tmp/profiled_data.html with your browser
+ * </ol>
+ */
+@SuppressWarnings("SystemOut")
+public class AsynchronousCounterMemoryUsageBenchmark {
```

Comment:
This class seems useful for exploration / validation. But given that its not runnable via JMH, and not runnable as part of test suites, it seems like it will just bit rot. I think we should delete it before merging.

Reply:
You are right. I kept a copy for me locally and deleted it.